sdks/java/core/src/main/java/org/apache/beam/sdk/values/reflect/DefaultRecordTypeFactory.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.sdk.values.reflect;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.math.BigDecimal;
+import java.util.List;
+import org.apache.beam.sdk.coders.BigDecimalCoder;
+import org.apache.beam.sdk.coders.BooleanCoder;
+import org.apache.beam.sdk.coders.ByteCoder;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.coders.DoubleCoder;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.coders.VarIntCoder;
+import org.apache.beam.sdk.coders.VarLongCoder;
+import org.apache.beam.sdk.values.BeamRecordType;
+import org.apache.beam.sdk.values.reflect.field.FieldValueGetter;
+
+/**
+ * A default factory to create a {@link BeamRecordType} based on pojo field getters.
+ */
+public class DefaultRecordTypeFactory implements RecordTypeFactory {
+
+  private static final ImmutableMap<Class, Coder> JAVA_TYPES_CODERS = ImmutableMap
```

Comment:
<!--thread_id:cc_154606380_t; commit:12389cee6a5061c472533ec063d041e03007687b; resolved:1-->
<!--section:context-quote-->
> **lukecwik** wrote:
> +1 on using the registry to figure this out.

<!--section:body-->
Changed to use `CoderRegistry.createDefault()`. Will update this to get the registry from the pipeline as soon as wire it up (future PRs).